extensions/azure/cosmos/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndex.java

```diff
@@ -105,6 +109,17 @@ public void accept(Asset asset, DataAddress dataAddress) {
         assetDb.saveItem(assetDocument);
     }
 
+    @Override
+    public Asset deleteById(String assetId) {
+        try {
+            var deletedItem = assetDb.deleteItem(assetId);
+            return convertObject(deletedItem).getWrappedAsset();
+        } catch (NotFoundException notFoundException) {
+            monitor.debug(String.format("Asset with id %s not found", assetId));
```

Comment:
```suggestion
            monitor.debug(() -> String.format("Asset with id %s not found", assetId));
```
use Supplier to save processing when debug is disabled.